### PR TITLE
Prevents ghosts from posessing megafauna

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
My guilt at seeing a colossus kill 30 out of 40 deathsquad members convinced me to make this PR I expect people to downvote it again until I give up but at least my I'll be morally in the clear until it happens again

### CODE REVIEW

**Comment by TechnoAlchemisto** [comment]
emergent gameplay

**Comment by Supermichael777** [comment]
cancerous as duck these things are deadly with snpc logic at the helm

**Comment by Zilenan91** [comment]
It IS wizard mode though so this is probably fine

**Comment by Niknakflak** [comment]
"Trying to fix an administration issue with a code solution"

**Comment by RemieRichards** [comment]
@Niknakflak only ghosts using the ```possess``` verb, from wizard rounds, doesn't affect admins even remotely (Admins can occupy the bosses, put other people in it etc. still)

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -538,6 +538,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 
 	if(!target)
 		return 0
+
+	if(istype (target, /mob/living/simple_animal/hostile/megafauna))
```
Comment (KazeEspada) [inline code/modules/mob/dead/observer/observer.dm]: SNOOOOOOOOOOOOOOWWWWWWWWWWWWWWWWWFLAKE

**Comment by InsaneHyena** [comment]
If it's a wizard round already, all kinds of ridiculous bullshit should be possible. Thumbs down.

File: code/modules/mob/dead/observer/observer.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/mob/dead/observer/observer.dm]: Do you have a better idea?

Comment (RemieRichards) [inline code/modules/mob/dead/observer/observer.dm]: @KorPhaeron A more generic solution, such as a var ```can_be_ghost_possessed```

Comment (KorPhaeron) [inline code/modules/mob/dead/observer/observer.dm]: Adding a var to every mob sounds worse though

Comment (RemieRichards) [inline code/modules/mob/dead/observer/observer.dm]: omg why is this attitude resurfacing? oranges has said it, now you've said it.

A GENERIC solution is var nicer than a snowflake ```istype()```
YES it's a little cluttered to have a var for every little thing but it's also infinitely more maintainable.

Comment (optimumtact) [inline code/modules/mob/dead/observer/observer.dm]: new var's add overhead to every var access on that type I remember @MrStonedOne saying that.

**Comment by Kromgar** [comment]
image

Please merge post haste

**Comment by PKPenguin321** [comment]
that image has convinced me that this PR should not be merged

**Comment by Kromgar** [comment]
image

**Comment by Kromgar** [comment]
image

**Comment by Kromgar** [comment]
image
image

**Comment by Kromgar** [comment]
image
image

**Comment by TheColdTurtle** [comment]
I liked it -the collosus

**Comment by optimumtact** [comment]
I'll be the bad guy

File: code/modules/mob/dead/observer/observer.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/mob/dead/observer/observer.dm]: what if I cheated and put some sort of faction or status flag check

Comment (optimumtact) [inline code/modules/mob/dead/observer/observer.dm]: stop overloading the faction var reee

**Comment by PKPenguin321** [comment]
o7

File: code/modules/mob/dead/observer/observer.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/mob/dead/observer/observer.dm]: @optimumtact I'm almost certain that's completely irrelevant with the var access upgrades for 511.

**Pull request merged**